prov/efa/src/rdm/efa_rdm_peer.h

```diff
@@ -128,6 +128,47 @@ bool efa_both_support_rdma_read(struct efa_rdm_ep *ep, struct efa_rdm_peer *peer
 	       (peer->is_self || efa_rdm_peer_support_rdma_read(peer));
 }
 
+/**
+ * @brief Determine interoperability for RDMA read between peers
+ *
+ * RDMA read is currently not interoperable between all EFA platforms; older
+ * platforms cannot fully interoperate with newer platforms. The platform
+ * version can be inferred from ibv_device_attr.vendor_part_id (see
+ * ibv_query_device(3)).
+ *
+ * In this context, version 0xEFA0 is considered "older", while versions 0xEFA1
+ * or 0xEFA2 are considered "newer"
+ *
+ * @todo Either:
+ * - Refactor this logic if the interoperability is ever reported
+ *   directly via firmware or rdma-core
+ * - Remove altogether if RDMA read becomes interoperable between all EFA
+ *   platforms
+ *
+ * @param[in]	ep	Endpoint for communication with peer
+ * @param[in]	peer	An EFA peer
+ *
+ * @return true if peers can interoperate via RDMA read; false otherwise
+ */
+static inline
+bool efa_rdm_interop_rdma_read(struct efa_rdm_ep *ep, struct efa_rdm_peer *peer)
+{
+	bool rdma_read_support = efa_both_support_rdma_read(ep, peer);
+	uint32_t ep_dev_ver = efa_rdm_ep_domain(ep)->device->ibv_attr.vendor_part_id,
+		 peer_dev_ver = peer->device_version;
+
+	switch (ep_dev_ver) {
+	case 0xEFA0:
```

Comment:
should we use lower case here to keep consistent with other places in the code for the vendor id.

Reply:
I opted for uppercase since that's what `ibv_devinfo` outputs for the `hw_ver` field (`vendor_part_id` is actually decimal for some reason). I don't feel strongly either way. Obviously it makes no difference functionally in this context.